app/services/order_creator.rb

```diff
@@ -0,0 +1,31 @@
+# frozen_string_literal: true
+
+require 'dry/monads'
+
+class OrderCreator
+  include Dry::Monads[:result, :do]
+
+  def call(employee:, reward:)
+    yield check_funds(employee, reward)
+    Success('Reward bought')
+  end
+
+  private
+
+  def check_funds(employee, reward)
+    if employee.points < reward.price
+      Failure('You have insufficient funds')
+    else
+      Success(create_order(employee, reward))
```

Comment:
this is imo a wrong use of monads because the context there lacks a yield that would unwrap a result. Although I planned to introduce service objects a bit later, it's ok to use them for now, but remove monads - they are a bit more advanced topic that we'll cover later in the course

Reply:
sure, understand, will fix this later today :) i can even go back "pre service object" if you want

Reply:
that would be cool because I'd like to show you guys why using SO is needed, but before we need to have few a bit nasty controllers we will refactor later :)